internal/controller/storageclient_controller.go

```diff
@@ -156,8 +210,77 @@ func (r *storageClientReconcile) reconcile() error {
 	return nil
 }
 
+func (r *storageClientReconcile) isCrdBeingWatched(crdName string) bool {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	return r.crdsBeingWatched[crdName]
+}
+
+func (r *storageClientReconcile) markCrdBeingWatched(crdName string) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	r.crdsBeingWatched[crdName] = true
+}
```

Comment:
You are just implementing SyncMap set and get operation here. Just another reason to use SyncMap